pkgs/stdenv/generic/builder.sh

```diff
@@ -6,15 +6,20 @@ done
 
 mkdir $out
 
+# Buid the setup script
 echo "export SHELL=$shell" > $out/setup
 echo "initialPath=\"$initialPath\"" >> $out/setup
-echo "defaultNativeBuildInputs=\"$defaultNativeBuildInputs\"" >> $out/setup
 echo "$preHook" >> $out/setup
 cat "$setup" >> $out/setup
 
 # Allow the user to install stdenv using nix-env and get the packages
 # in stdenv.
-mkdir $out/nix-support
+mkdir -p "$out/nix-support"
+echo '# Hack to induce runtime dependencies on the default inputs' \
```

Comment:
The nixos-tests copy the stdenv runtime closure to the vm, so it's important the runtime closure contain everything it did before.